purchase_discount/models/purchase_order.py

```diff
@@ -61,20 +61,8 @@ def _get_stock_move_price_unit(self):
         """Get correct price with discount replacing current price_unit
         value before calling super and restoring it later for assuring
         maximum inheritability.
-
-        HACK: This is needed while https://github.com/odoo/odoo/pull/29983
-        is not merged.
         """
-        price_unit = False
-        price = self._get_discounted_price_unit()
-        if price != self.price_unit:
-            # Only change value if it's different
-            price_unit = self.price_unit
-            self.price_unit = price
-        price = super()._get_stock_move_price_unit()
-        if price_unit:
-            self.price_unit = price_unit
-        return price
+        return self._get_discounted_price_unit()
```

Comment:
Only possible when this gets merged: https://github.com/odoo/odoo/pull/92933/files

Otherwise, you should provide a context to avoid recurssion

Reply:
Ok, I've added a context to avoid recursion.